File: 0110-balanced-binary-tree/0110-balanced-binary-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
 #include<stdlib.h>
 int max(int l, int r){
    if(l>r){
        return l;
    }
    return r;
 }
 int height(struct TreeNode* root)
 {
    if(root==NULL){
        return 0;
    }
    int left=height(root->left);
    
    int right=height(root->right);
    if(right==-1){
        return -1;
    }
    else if(left==-1){
        return -1;
    }
    else if(abs(left-right)>1){
        return -1;

    }
    else{
        return 1+max(left,right);
    }


 }
bool isBalanced(struct TreeNode* root) {
    int x=height(root);
    if(x==-1){
        return false;
    }
    else{
        return true;
    }

    
}

File: 0110-balanced-binary-tree/0110-balanced-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        def height(root):
            if root is None:
                return True
            left=height(root.left)
            right=height(root.right)
            if left is -1:
                return -1
            elif right is -1:
                return -1
        
            elif abs(left-right)>1:
                 return -1
            else:
                 return max(left,right)+1 
        x=height(root)
        if x is -1:
            return False
        else:
            return True

        